fdbserver/workloads/Mako.actor.cpp

```diff
@@ -0,0 +1,530 @@
+							uint8_t *keyPtr = mutateString(tempKey);
+							randStr((char*) keyPtr, self->keyBytes-KEYPREFIXLEN);
+							for (int range_i = 0; range_i < range; ++range_i){
+								sprintf((char*) keyPtr + self->keyBytes - KEYPREFIXLEN - rangeLen, "%0.*d", rangeLen, range_i);
+								tr.set(tempKey, self->randomValue(), true);
+							}
+							doCommit = true;
+						} else if (i == OP_CLEAR){
+							tr.clear(rkey, true);
+							doCommit = true;
+						} else if(i == OP_SETCLEAR){
+							state std::string st_key;
+							randStr(st_key, self->keyBytes-KEYPREFIXLEN);
+							tr.set(StringRef(st_key), rval, true);
+							// commit the change and update metrics
+							commitStart = now();
+							wait(tr.commit());
+							self->opLatencies[OP_COMMIT].addSample(now() - commitStart);
+							++perOpCount[OP_COMMIT];
+							tr.reset();
+							tr.clear(st_key, true);
+							doCommit = true;
+						} else if (i == OP_CLEARRANGE){
+							tr.clear(rkeyRangeRef, true);
+							doCommit = true;
+						} else if (i == OP_SETCLEARRANGE){
+							// Key tempKey[self->keyBytes-KEYPREFIXLEN+1];
+							Key tempKey = makeString(self->keyBytes-KEYPREFIXLEN);
+							uint8_t *keyPtr = mutateString(tempKey);
+							randStr((char*) keyPtr, self->keyBytes-KEYPREFIXLEN);
+							state std::string scr_start_key;
+							state std::string scr_end_key;
+							for (int range_i = 0; range_i < range; ++range_i){
+								sprintf((char*) keyPtr + self->keyBytes - KEYPREFIXLEN - rangeLen, "%0.*d", rangeLen, range_i);
+								tr.set(tempKey, self->randomValue(), true);
+								if (range_i == 0)
+									scr_start_key = tempKey.toString();
+							}
+							scr_end_key = tempKey.toString();
+							commitStart = now();
+							wait(tr.commit());
+							self->opLatencies[OP_COMMIT].addSample(now() - commitStart);
+							++perOpCount[OP_COMMIT];
+							tr.reset();
+							tr.clear(KeyRangeRef(StringRef(scr_start_key), StringRef(scr_end_key)), true);
+							doCommit = true;
+						}
+						++perOpCount[i];
+					}
+				}
+
+				if (doCommit) {
+					commitStart = now();
+					wait(tr.commit());
+					self->opLatencies[OP_COMMIT].addSample(now() - commitStart);
+					++perOpCount[OP_COMMIT];
+				}
+				// successfully finish the transaction, update metrics
+				++self->xacts;
+				for (int op = 0; op < MAX_OP; ++op){
+					self->opCounters[op] += perOpCount[op];
+					self->totalOps += perOpCount[op];
+				}
+			} catch (Error& e) {
+				// TODO: to see how to log the error here
+				if (e.code() == error_code_operation_cancelled)
+					throw;
+				else if (e.code() == error_code_not_committed)
+					++self->conflicts;
+				
+				wait(tr.onError(e));
+				++self->retries;
+			}
+			// reset all the operations' counters to 0
+			std::fill(perOpCount.begin(), perOpCount.end(), 0);
+			tr.reset();
+		}
+	}
+
+	ACTOR Future<Void> cleanup(Database cx, MakoWorkload* self){
+		// clear all data in the database
+		state std::string startKey("");
```

Comment:
I clear all since if the transactions have `insert` related operations. 
To make inserted ones unique, we do not add the prefix to them.
I clear all since there are keys which do not start with `mako`.